python/tchannel/messages/call_request.py

```diff
@@ -33,42 +28,65 @@ def __init__(
         service=None,
         headers=None,
         checksum=None,
-        arg_1=None,
-        arg_2=None,
-        arg_3=None,
+        args=None,
     ):
-        self.flags = flags
+        super(CallRequestMessage, self).__init__(flags, checksum, args)
         self.ttl = ttl
         self.tracing = tracing or common.Tracing(0, 0, 0, 0)
         self.service = service or ''
         self.headers = dict(headers) if headers else {}
-        if checksum is not None:
-            checksum = common.ChecksumType.standardize(checksum)
-        self.checksum = checksum or \
-            (common.ChecksumType.none, None)
 
-        self.arg_1 = arg_1 or ''
-        self.arg_2 = arg_2 or ''
-        self.arg_3 = arg_3 or ''
+    def encode(self):
+        super(CallRequestMessage, self).encode()
+        self.service = self.service.encode(common.ENCODE_TYPE)
```

Comment:
`service` is expected to be a UTF-8 string and is automatically encoded/decoded by `len_prefixed_string`.